packages/patternfly-4/react-charts/src/components/ChartDonut/ChartDonut.js

```diff
@@ -1,21 +1,164 @@
 import { VictoryPie } from 'victory';
-import { default as ChartTheme } from '../ChartTheme/ChartTheme';
+import ChartLabel from '../ChartLabel/ChartLabel';
+import { styles } from '../ChartTheme/themes/theme-donut';
+import { getDonutTheme } from '../ChartTheme/themes/theme-utils';
 import ChartTooltip from '../ChartTooltip/ChartTooltip';
 
 export const propTypes = {
   /**
    * See TypeScript API docs: https://formidable.com/open-source/victory/docs/victory-area/
    */
-  '': PropTypes.any
+  '': PropTypes.any,
+  /**
+   * The height props specifies the height the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  height: PropTypes.number,
+  /**
+   * The subtitle for the donut chart
+   */
+  subTitle: PropTypes.string,
+  /*
+   * Specifies the theme color; blue (default), green, or multi-color. Overridden by the theme property.
+   */
+  themeColor: PropTypes.string,
+  /*
+   * Specifies the theme variant; 'dark' or 'light' (default). Overridden by the theme property.
+   */
+  themeVariant: PropTypes.string,
+  /**
+   * The title for the donut chart
+   */
+  title: PropTypes.string,
+  /**
+   * The width props specifies the width of the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number
 };
 
+class ChartDonut extends React.Component {
+  titleRef = React.createRef();
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      height: this.props.height || 0,
+      width: this.props.width || 0
+    };
+  }
+
+  componentDidMount() {
+    // Skip resize handler if height and width were provided
+    if (!(this.state.height || this.state.width)) {
+      setTimeout(() => {
+        window.addEventListener('resize', this.handleResize);
+      });
+      this.handleResize();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleResize);
+  }
+
+  getSubTitle = () => {
+    const { subTitle } = this.props;
+    const { height, width } = this.state;
+
+    if (subTitle && height && width) {
+      const xVal = width / 2 - subTitle.length / 2;
```

Comment:
Isn't 'subTitle.length' the number of chars in the string? Since `width` is pixels, I could see this causing alignment issues. Also, if this were wrapped in an svg with the `VictoryPie` component, they will share the same coordinate system and the svg textAnchor prop should take care of centering the text horizontally.

Reply:
textAnchor defines how the text is horizontally positioned relative to the given x and y coordinates. It's not perfect, and may iterate on this, but need to get x and y close for now.